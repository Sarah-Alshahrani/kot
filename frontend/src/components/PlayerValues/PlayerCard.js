import React from "react";
import Info from "./../Images/icons8-info-128.png";
import Energy from "./../Images/Energy.png";
import Footnote from "./../Images/icons8-footsteps-50.png";
import "./PlayerCard.css";

class PlayerCard extends React.Component {
  render() {
    let t_footnote = "None";
    if (this.props.footnote) {
      t_footnote = this.props.footnote;
    }

    return (
      <div>
        <div className="card-border">
          <p className="card-instances">
            {this.props.type}
            {this.props.name}
            <br />
            <span>
              {this.props.cost}
              <img className="img-energy" alt="" src={Energy}></img>
            </span>
            <span data-tooltip={this.props.effect}>
              <img className="img-info" alt="" src={Info}></img>
            </span>
            <span data-tooltip={t_footnote}>
              <img className="img-info" alt="" src={Footnote}></img>
            </span>
          </p>
        </div>
      </div>
    );
  }
}
export default PlayerCard;
