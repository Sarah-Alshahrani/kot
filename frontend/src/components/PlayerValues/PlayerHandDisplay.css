.player_hand_details {
    justify-content: center;
    display: flex;
    flex-direction: column;
    border-width: thin
}

.img-hand {
    margin: 3px;
    height: 55px
}