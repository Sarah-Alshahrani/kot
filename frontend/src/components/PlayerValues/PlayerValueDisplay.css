.player_details {
    justify-content: center;
    display: flex;
    flex-direction: row;
    border-width: thin
}