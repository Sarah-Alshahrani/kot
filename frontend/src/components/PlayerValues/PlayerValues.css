.player-border {
    border: 2px solid;
    border-radius: 20px
}

.player-values {
    margin: 7px;
    font-family: 'Open Sans', sans-serif;
    font-size: 22px;
    text-align: left;
    color: black;
}
img {
    margin: 3px;
    height: 15px    
}