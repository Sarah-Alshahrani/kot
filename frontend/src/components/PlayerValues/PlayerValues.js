import React from "react";
import Heart from "./../Images/Heart.gif";
import Star from "./../Images/Star.png";
import Energy from "./../Images/Energy.png";
import "./PlayerValues.css";

class PlayerValues extends React.Component {
  render() {
    return (
      <div>
        <div className="player-border">
          <h2>Player {this.props.username}</h2>
          <div className="player-values">
            <p>
              Victory Points:
              <span className="badge badge-light">
                {this.props.victory_points}
              </span>
              <img alt="" src={Star}></img>
            </p>
            <p>
              Health:
              <span className="badge badge-light">
                {this.props.current_health}
              </span>
              <img alt="" src={Heart}></img>
            </p>
            <p>
              Energy:
              <span className="badge badge-light">{this.props.energy}</span>
              <img alt="" src={Energy}></img>
            </p>
            <p>
              In/Out of Tokyo:
              <span className="badge badge-warning">{this.props.location}</span>
            </p>
          </div>
        </div>
      </div>
    );
  }
}
export default PlayerValues;
