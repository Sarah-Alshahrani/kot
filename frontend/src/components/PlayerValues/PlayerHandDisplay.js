import React from "react";
import Hand from "./../Images/icons8-red-card-80.png";
import "./PlayerHandDisplay.css";
import PlayerCard from "./PlayerCard";

class PlayerHandDisplay extends React.Component {
  constructor(props, context) {
    super(props, context);

    let cards = this.updateCards(this.props.data);
    this.state = {
      username: this.props.currentUser,
      gameRoom: this.props.currentRoom,
      data: cards
    };
  }

  updateCards(data) {
    let cards = {};
    if (this.props.data) {
      try {
        cards = JSON.parse(data);
      } catch (e) {}
    }
    return cards;
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.data !== prevProps.data) {
      let cards = this.updateCards(this.props.data);
      this.setState({
        data: cards
      });
    }
  }

  render() {
    if (this.state.data && this.state.data.length > 0) {
      return (
        <div>
          <img className="img-hand" alt="" src={Hand}></img>
          <container className="player_hand_details">
            {this.state.data.map((entry, index) => (
              <PlayerCard
                key={index}
                name={entry.name}
                cost={entry.cost}
                effect={entry.effect}
                footnote={entry.footnote}
              />
            ))}
          </container>
        </div>
      );
    } else {
      return (
        <div>
          <img className="img-hand" alt="" src={Hand}></img>
          No Cards...
        </div>
      );
    }
  }
}
export default PlayerHandDisplay;
