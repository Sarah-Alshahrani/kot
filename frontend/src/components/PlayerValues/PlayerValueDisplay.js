import React from "react";
import "./PlayerValueDisplay.css";
import PlayerValues from "./PlayerValues";
import GameInstance from "../../services/gameService";

import MT_RESPONSE_PLAYERS_STATUS_UPDATE from "../../services/config";

class PlayerValueDisplay extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      username: props.currentUser,
      gameRoom: props.currentRoom,
      data: props.data,
      justShowSelf: props.displayOnlySelf
    };

    GameInstance.addCallback(
      MT_RESPONSE_PLAYERS_STATUS_UPDATE,
      this.playerUpdateHandler.bind(this)
    );
  }

  playerUpdateHandler(message) {
    const content = message.content;

    if (content === undefined || content === "") return;

    let player_status = [];
    try {
      player_status = JSON.parse(content);
    } catch (e) {}
    this.setState({ data: player_status });
  }

  render() {
    if (this.state.data) {
      let you = this.state.username;
      let justSelf = this.state.justShowSelf;
      let items = [];
      this.state.data.forEach(element => {
        if (justSelf && element.username === you) {
          items.push(element);
        } else if (!justSelf && element.username !== you) {
          items.push(element);
        }
      });

      if (items && items.length) {
        let tisYou = this.state.username;

        return (
          <container className="player_details">
            {items.map((entry, index) => (
              <PlayerValues
                key={index}
                username={entry.username}
                thisIsYou={entry.username === tisYou}
                victory_points={entry.victory_points} // 0 to 10
                current_health={entry.current_health} // 0 to 10
                energy={entry.energy} // 0 or more
                location={entry.location} // 'In' or 'Out'
                cards={entry.cards}
              />
            ))}
          </container>
        );
      }
    }

    return <div>Waiting on Players</div>;
  }
}
export default PlayerValueDisplay;
