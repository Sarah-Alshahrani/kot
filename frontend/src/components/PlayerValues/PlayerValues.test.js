import React from "react";
import { shallow } from "enzyme";

import PlayerValues from "./PlayerValues";

describe("Verify player value template", () => {
  it("Player values should be rendered correctly", () => {
    const errorPage = shallow(<PlayerValues />);
    expect(errorPage).toMatchSnapshot();
  });

  describe("<PlayerValues />", () => {
    const expected_player_name = "player_name";
    const expected_victory_points = "victory_points";
    const expected_health = "health";
    const expected_energy = "energy";
    const expected_in_or_out_position = "in_or_out_position";

    const testValues = {
      player_name: expected_player_name,
      victory_points: expected_victory_points, // 0 to 10
      health: expected_health, // 0 to 10
      energy: expected_energy, // 0 or more
      in_or_out_position: expected_in_or_out_position // 'In' or 'Out'
    };

    it("Submit works", () => {
      const wrapper = shallow(
        <PlayerValues
          player_name={testValues.player_name}
          victory_points={testValues.victory_points} // 0 to 10
          health={testValues.health} // 0 to 10
          energy={testValues.energy} // 0 or more
          in_or_out_position={testValues.in_or_out_position} // 'In' or 'Out'
        />
      );
      expect(wrapper.state()).toEqual({ ...testValues }); // passed
    });
  });
});
