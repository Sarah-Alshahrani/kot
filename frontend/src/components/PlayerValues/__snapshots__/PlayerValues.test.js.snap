// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify player value template Player values should be rendered correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm player-border"
      >
        <h2 />
        <p>
          them
        </p>
        <div
          className="player-values"
        >
          <p>
            Victory Points:
            <span
              className="badge badge-light"
            />
            <img
              alt=""
              src="Star.png"
            />
          </p>
          <p>
            Health:
            <span
              className="badge badge-light"
            />
            <img
              alt=""
              src="Heart.gif"
            />
          </p>
          <p>
            Energy:
            <span
              className="badge badge-light"
            />
            <img
              alt=""
              src="Energy.png"
            />
          </p>
          <p>
            In/Out of Tokyo:
            <span
              className="badge badge-warning"
            />
          </p>
        </div>
      </div>
      <div
        className="col-sm"
      >
        <PlayerHandDisplay />
      </div>
    </div>
  </div>
</div>
`;
